python/tank/deploy/tank_commands/core_upgrade.py

```diff
@@ -162,88 +174,90 @@ def _run(self, log, suppress_prompts):
         return return_status
-    
 
 
-class TankCoreUpgrader(object):
+class TankCoreUpdater(object):
     """
-    Class which handles the upgrade of the core API.
+    Class which handles the update of the core API.
     """
     
     # possible update status states
     (UP_TO_DATE,                    # all good, no update necessary  
-     UPGRADE_POSSIBLE,              # more recent version exists
-     UPGRADE_BLOCKED_BY_SG          # more recent version exists but SG version is too low.
+     UPDATE_POSSIBLE,              # more recent version exists
+     UPDATE_BLOCKED_BY_SG          # more recent version exists but SG version is too low.
      ) = range(3)
-        
-        
-    def __init__(self, installation_root, logger):
+
+    def __init__(self, installation_root, logger, core_version):
         """
         Constructor
-        
+
         :param installation_root: The path to the installation to check. This is either a localized
                                   Pipeline Configuration or a studio code location (omit the install folder).
                                   Because we are passing this parameter in explicitly, the currently running
-                                  code base does not have to be related to the code base that is being upgraded,
-                                  e.g. you can run the upgrader as a totally separate thing.
+                                  code base does not have to be related to the code base that is being updated,
+                                  e.g. you can run the updater as a totally separate thing.
         :param logger: Logger to send output to.
+        :param core_version: Version of the core to update to. If None, the core will be updated to the latest
+                             version.
         """
         self._log = logger
-        
+
         (sg_app_store, script_user) = shotgun.create_sg_app_store_connection()
         self._sg = sg_app_store
         self._sg_script_user = script_user
-        
-        self._local_sg = shotgun.get_sg_connection()      
-        self._latest_ver = self.__get_latest_version()
-        
+
+        self._local_sg = shotgun.get_sg_connection()
+        self._update_version = self.__get_core_version(core_version)
+
         self._install_root = os.path.join(installation_root, "install")
-        
+
         self._current_ver = pipelineconfig_utils.get_currently_running_api_version()
-         
+
         # now also extract the version of shotgun currently running
         try:
-            self._sg_studio_version = ".".join([ str(x) for x in self._local_sg.server_info["version"]])        
+            self._sg_studio_version = ".".join([ str(x) for x in self._local_sg.server_info["version"]])
         except Exception, e:
             raise TankError("Could not extract version number for studio shotgun: %s" % e)
-    
-            
-    def __get_latest_version(self):
+
+    def __get_core_version(self, core_version):
```

Comment:
i think we should do this change off the `develop/0.18` branch, which supports core descriptors. then we can just the descriptor framework to resolve the code rather than having to hard code all this stuff. We would also get the descriptor based core syntax pretty much for free (e.g we get both `./tank core --version=v1.2.3`
and `./tank core --version="sgtk:descriptor:dev?path=/Users/manne/Documents/work_dev/toolkit/tk-core"`)